src/shared/components/community/communities.tsx

```diff
@@ -221,7 +221,14 @@ export class Communities extends Component<any, CommunitiesState> {
                 </tbody>
               </table>
             </div>
-            <Paginator page={page} onChange={this.handlePageChange} />
+            <Paginator
+              page={page}
+              onChange={this.handlePageChange}
+              nextDisabled={
+                communityLimit >
+                this.state.listCommunitiesResponse.data.communities.length
```

Comment:
Looks good. Could you add this logic also everywhere else that its feasible? (IE Home.tsx, Community.tsx especially) 

The pages with multiple request types (like `inbox.tsx`, `modlog.tsx` , it won't work, so you can ignore those ones)